src/lib.rs

```diff
@@ -0,0 +1,497 @@
+                let rsa_options = Some(RsaOptions {
+                    signature_policy: vec![sig_policy],
+                    ..Default::default()
+                });
+
+                SobjectRequest {
+                    name: Some(name),
+                    description,
+                    obj_type: Some(ObjectType::Rsa),
+                    key_ops: Some(ops),
+                    key_size: Some(*key_size),
+                    rsa: rsa_options,
+                    ..Default::default()
+                }
+            }
+            (KeyRole::Subkey, SupportedPkAlgo::Rsa(key_size)) => {
+                let ops = KeyOperations::DECRYPT | KeyOperations::APPMANAGEABLE;
+                let enc_policy = RsaEncryptionPolicy {
+                    padding: Some(RsaEncryptionPaddingPolicy::Pkcs1V15 {}),
+                };
+                let rsa_options = Some(RsaOptions {
+                    encryption_policy: vec![enc_policy],
+                    ..Default::default()
+                });
+                let name = name + " (PGP: decryption subkey)";
+
+                SobjectRequest {
+                    name: Some(name),
+                    description,
+                    obj_type: Some(ObjectType::Rsa),
+                    key_ops: Some(ops),
+                    key_size: Some(*key_size),
+                    rsa: rsa_options,
+                    ..Default::default()
+                }
+            }
+        };
+
+        let sobject = client.create_sobject(&sobject_request)?;
+
+        PublicKey::from_sobject(sobject, role)
+    }
+
+    fn from_sobject(sob: Sobject, role: KeyRole) -> Result<Self> {
+        let des = SobjectDescriptor::Kid(sob.kid.context("no kid in sobject")?);
+        let time = sob.created_at.to_datetime();
+        let raw_pk =
+            sob.pub_key.context("public bits of sobject not returned")?;
+
+        match sob.obj_type {
+            ObjectType::Rsa => {
+                let deserialized_pk = Pk::from_public_key(&raw_pk).context(
+                    "cannot deserialize SDKMS key into mbedTLS object",
+                )?;
+
+                let (e, n) = (
+                    deserialized_pk.rsa_public_exponent()?.to_be_bytes(),
+                    deserialized_pk.rsa_public_modulus()?.to_binary()?,
+                );
+                let key = Key::V4(
+                    Key4::import_public_rsa(&e, &n, Some(time.into()))
+                        .context("cannot import RSA key into Sequoia object")?,
+                );
+
+                Ok(PublicKey {
+                    descriptor: des,
+                    role,
+                    sequoia_key: Some(key),
+                })
+            }
+            _ => unimplemented!(),
+        }
+    }
+
+    fn uid(&self) -> Result<Uuid> {
+        match self.descriptor {
+            SobjectDescriptor::Kid(x) => Ok(x),
+            _ => Err(anyhow::Error::msg("bad descriptor")),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests;
```

Comment:
Remove this line?

Reply:
The `tests` module declaration? If I do so, `cargo test` won't see the tests

Reply:
Oh right, my bad.